Lesson16/bilge/script.js

```diff
@@ -0,0 +1,416 @@
+    return true;
+  } else {
+    return false;
+  }
+}
+console.log(isFirstLetterUpperCase(""));
+console.log(isFirstLetterUpperCase("Write a function that"));
+
+// 5. Write a function that takes in a number and checks if it's positive, negative, or zero using if else statement.
+
+function checkPositiveOrNegative(conditionalNumber) {
+  if (conditionalNumber < 0) {
+    return "The number is negative.";
+  } else if (conditionalNumber === 0) {
+    return "The number is zero.";
+  } else {
+    return "The number is positive.";
+  }
+}
+
+console.log('Ex. 5:');
+
+console.log(checkPositiveOrNegative(-5));
+console.log(checkPositiveOrNegative(20));
+console.log(checkPositiveOrNegative(0));
+
+// 6. Write a function that takes in a year and checks if it's a leap year (is divisible by 4 and not divisible by 100 OR divisible by 400) or not using if else statement.
+
+function isItLeapYear(year) {
+  if (year % 4 === 0 && year % 100 != 0 || year % 400 === 0) {
+    return "It's a leap year.";
+  } else {
+    return "It's not a leap year.";
+  }
+}
+
+console.log('Ex. 6:');
+
+console.log(isItLeapYear(2024));
+console.log(isItLeapYear(1979));
+
+// 7. Write a function that takes in a temperature and checks if it's above or below freezing using if else statement.
+
+function isItAboveOrBelowFreezing (temp) {
+  if (temp < 0) {
+    return 'It\'s below freezing.'
+  }
+  else if (temp === 0) {
+    return 'It\'s the freezing point.'
+  }
+  else {
+    return 'It\'s above freezing.'
+  }
+}
+
+console.log('Ex. 7:');
+
+console.log(isItAboveOrBelowFreezing(-1));
+console.log(isItAboveOrBelowFreezing(25));
+console.log(isItAboveOrBelowFreezing(0));
+
+// 8. Write a function that takes in a person's height and weight and checks if they are considered underweight, normal, overweight, or obese using if else statement.
+// Body Mass Index (BMI) is a person’s weight in kilograms (or pounds) divided by the square of height in meters (or feet). A high BMI can indicate high body fatness.
+//Formula: weight (kg) / [height (m)]2
+// Below 18.5	Underweight
+// 18.5 – 24.9	Healthy Weight
+// 25.0 – 29.9	Overweight
+// 30.0 and Above	Obesity
+
+function findBodyMassIndex (weight, height) {
+  let bmi = weight / (height ** 2); 
+
+  if (bmi < 18.5) {
+    return 'The person is underweight.'
+  }
+  else if (bmi >= 18.5 && bmi <= 24.9) {
+    return 'The person is normal.'
+  }
+  else if (bmi >=25 && bmi <= 29.9) {
+    return 'The person is overweight.'
+  }
+  else {
+    return 'The person is obese.'
+  }
+}
+
+console.log('Ex. 8:');
+
+console.log(findBodyMassIndex(40, 1.70));
+console.log(findBodyMassIndex(60, 1.70));
+console.log(findBodyMassIndex(80, 1.70));
+console.log(findBodyMassIndex(200, 1.70));
+
+// 9. Write a function that takes in a person's age and checks if they are a child (age 0-12), teenager (age 13-19), adult (age 20-64), or senior citizen (age 65 and above) using if else statement.
+
+function checkAge(ageOfThePerson) {
```

Comment:
It's enough to name parameter 'age' in this case.
```suggestion
function checkAge(age) {
```

Reply:
This makes more sense!